programs/fileio.c

```diff
@@ -769,6 +769,15 @@ static unsigned FIO_highbit64(unsigned long long v)
     return count;
 }
 
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+
+static void FIO_adjustMemLimitForPatchFromMode(FIO_prefs_t* const prefs,
+                                    unsigned long long const dictSize,  
+                                    unsigned long long const maxSrcFileSize)
+{
+    if (dictSize != UTIL_FILESIZE_UNKNOWN && maxSrcFileSize != UTIL_FILESIZE_UNKNOWN)
+        FIO_setMemLimit(prefs, MAX(prefs->memLimit, MAX((unsigned)dictSize, (unsigned)maxSrcFileSize)));
```

Comment:
Question :
what happens when `dictSize` or `maxSrcFileSize` is > 4 GB ?

It seems this invocation could suffer an overflow here,
unless a limit range condition is checked elsewhere
in which case it should be an `assert()` here.

Reply:
It should throw an error now